### PR TITLE
fix: Adding validation on kms plaintext based on the key type 

### PR DESCRIPTION
**Pull request by jcanabarro**
The issue was partially fixed here #8083, but I also need to check the context based on the key type, not only the max allowed plaintext size.
* Added method to validate the plaintext size based on the key type, all those extracted numbers are from [boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms/client/encrypt.html)
* Add tests cases to the written function

### CODE REVIEW

**Comment by jcanabarro** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jcanabarro** [comment]
recheck

**Comment by jcanabarro** [comment]
recheck

**Comment by viren-nadkarni** [comment]
@jcanabarro Some of your earlier commits are associated with `joao@querylayer.com`. Could you squash/amend all you commits and use `joaovictorcana@gmail.com` for all the commits?

File: localstack/services/kms/provider.py [hunk 1]
```diff
@@ -1018,6 +1019,33 @@ def _validate_grant_request(self, data: Dict, store: KmsStore):
...
+    def _validate_plaintext_key_type_based(
+        self,
+        plaintext: PlaintextType,
+        key: KmsKey,
+        encryption_algorithm: EncryptionAlgorithmSpec = None,
+    ):
+        # max size values extracted from AWS boto3 documentation
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms/client/encrypt.html
+        max_size_bytes = 4096  # max allowed size
+        if key.metadata["KeySpec"] == "RSA_2048" and encryption_algorithm == "RSAES_OAEP_SHA_1":
```
Comment (sannya-singal) [inline localstack/services/kms/provider.py:1031]: We can use the key spec values from https://github.com/localstack/localstack/blob/ed7e02f18253e22780131cf961304aa9e9df5c85/localstack/aws/api/kms/__init__.py#L158 and for encryption algorithm https://github.com/localstack/localstack/blob/ed7e02f18253e22780131cf961304aa9e9df5c85/localstack/aws/api/kms/__init__.py#L122.

File: tests/integration/test_kms.py [hunk 2]
```diff
@@ -542,6 +542,28 @@ def test_encrypt_decrypt(self, kms_create_key, key_spec, algo, aws_client):
         )["Plaintext"]
         assert base64.b64decode(plaintext) == message
 
+    @pytest.mark.parametrize(
```
Comment (sannya-singal) [inline tests/integration/test_kms.py:545]: We can add `pytest.mark.aws_validated` handler if it tested against AWS.

**Review by sannya-singal — CHANGES_REQUESTED** [review]
Thanks for adding this validation 🎉.

**Comment by jcanabarro** [comment]
@sannya-singal I'll abandon the PR and recreate because even with the rebase I'm still facing issues with the CLA.

**Pull request closed without merging**